### PR TITLE
Remove builders from Config(s) and examples.

### PR DESCRIPTION
**Pull request by bobmcwhirter**

### CODE REVIEW

File: examples/stm32h7/src/bin/dac.rs [hunk 1]
```diff
@@ -54,3 +52,16 @@ fn to_sine_wave(v: u8) -> u8 {
...
+    config.rcc = rcc_config();
+    config
+}
+
+fn rcc_config() -> rcc::Config {
+    let mut config = rcc::Config::default();
+    config.sys_ck = Some(400.mhz().into());
+    config.pll1.q_ck = Some(100.mhz().into());
+    config
+}
```
Comment (timokroeger) [inline examples/stm32h7/src/bin/dac.rs:67]: Duplicated for each h7 example. Could be moved to `example_common`

**Comment by bobmcwhirter** [comment]
Good point!

File: examples/stm32h7/src/bin/dac.rs (hunk 1 above)
Comment (Dirbaio) [reply examples/stm32h7/src/bin/dac.rs:67]: I would make it a single method too. `config.rcc.sys_ck = xx`

File: examples/stm32h7/src/bin/eth.rs [hunk 2]
```diff
@@ -108,16 +108,11 @@ fn main() -> ! {
...
-    rcc_config.pll1.q_ck = Some(Hertz(100_000_000));
-    let config = Config::default().rcc(rcc_config);
-
-    let mut p = embassy_stm32::init(config);
+    let mut p = embassy_stm32::init(config());
 
     // Constrain and Freeze clock
 
-    let mut rcc = Rcc::new(&mut p.RCC, RccConfig::default());
+    let mut rcc = Rcc::new(&mut p.RCC, rcc::Config::default());
```
Comment (Dirbaio) [inline examples/stm32h7/src/bin/eth.rs:115]: This is initializing RCC twice (though it already did that before...?), it shouldn't do so :S

File: examples/stm32f0/src/example_common.rs [hunk 3]
```diff
@@ -12,7 +12,9 @@ use embassy_stm32::Config;
 pub fn config() -> Config {
     let mut rcc_config = rcc::Config::default();
     rcc_config.enable_debug_wfe = true;
-    Config::default().rcc(rcc_config)
+    let mut config = Config::default();
+    config.rcc = rcc_config;
```
Comment (Dirbaio) [inline examples/stm32f0/src/example_common.rs:16]: `config.rcc.enable_debug_wfe = true`, so you don't need two vars.

**Review by Dirbaio — APPROVED** [review]
:rocket: Thank you!

**Pull request merged**